Test user db

yajaneya opened the pull request:
Тесты добавления и удаления пользователя в локальной БД
https://trello.com/c/Wl8uGbIK/115-%D1%81%D0%BE%D0%B7%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5-%D1%82%D0%B5%D1%81%D1%82%D0%B0-%D0%B4%D0%BE%D0%B1%D0%B0%D0%B2%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F-%D0%B8-%D1%83%D0%B4%D0%B0%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F-%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D0%B5%D0%BB%D1%8F-%D0%B2-%D0%B1%D0%B4

yajaneya committed:
Merge pull request #2 from siblis/develop

Develop

yajaneya committed:
Merge pull request #3 from siblis/develop

Develop

yajaneya committed:
Тест БД по добавлению и удалению пользователя.

yajaneya committed:
Добавление junit в maven

yajaneya committed:
Добавлена проверка на наличие пользователя с тестовым id в БД и изменение id в этом случае.
Добавлена проверка на наличие тестового пользователя в конце теста и удаление его в этом случае.

yajaneya committed:
Добавлены проверки на наличие пользователя в начале и отсутствие его в конце, и соотвтетсвующие действия.

yajaneya committed:
Сделал поле ID приватным

yajaneya committed:
Merge pull request #4 from siblis/develop

Develop

yajaneya committed:
Merge branch 'develop' into testUserDB

yajaneya committed:
Добавил создание тестовой БД

PaLoMaster commented on pom.xml:
```diff
@@ -122,6 +122,18 @@
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+        </dependency>
```
Лишняя зависимость при правильном расположении тестового файла.

PaLoMaster commented on src/main/java/tests/testUserDB.java:
```diff
@@ -0,0 +1,52 @@
+package tests;
```
1. Файл должен быть в папке src\test\java
2. Класс должен быть в пакете ru.geekbrains.pocket.messenger.database.dao

PaLoMaster commented on src/main/java/tests/testUserDB.java:
```diff
@@ -0,0 +1,52 @@
+    @After
+    public void finish() {
+        if (!(dbs.getUserById(Id)==null))
+            dbs.deleteUser(user);
```
Надо бы ещё и сам файл БД удалить.

PaLoMaster reviewed (commented):
На сколько понимаю (основываясь на своих малых знаниях процесса тестирования в maven)...
Тестовый класс (и файл) должен называться DataBaseServiceTest, имена тестов - типа testMethod (например, testInsertUser для метода insertUser). В присланном тобой примере была подсказка, что idea может сама создать тест для класса - не пробовал?
Перед вставкой и после удаления не хватает проверки на null (что пользователя нет в БД), а перед удалением - проверки на его присутствие в БД. Хотя, на сколько знаю, тестовые методы могут выполняться параллельно и в связи с этим у меня возникает вопрос о монопольном доступе к dbs - надо покурить мануалы...

PaLoMaster commented on src/main/java/tests/testUserDB.java:
```diff
@@ -0,0 +1,52 @@
+        while (!(w==null)) {
+            Id = (++id).toString();
+            w = dbs.getUserById(Id);
+        }
```
```
        User w;
        while ((w = dbs.getUserById(Id)) != null) {
            Id = (++id).toString();
        }
``` 
:)

yajaneya committed:
Перенес тест в директорию scr/test.
Добавил удаление тестовой БД в конце теста.
Убрал удаление пользователя, так как он удаляется с БД.

yajaneya committed:
Доп.комит - остальные изменения

ele-o-nora commented on src/test/ru/geekbrains/pocket/messenger/database/dao/DataBaseServiceTest.java:
```diff
@@ -1,5 +1,53 @@
+    public void deleteUser() {
+        dbs.insertUser(user);
+        dbs.deleteUser(user);
+        Assert.assertNotEquals(user, dbs.getUserById(Id));
```
Мне кажется, здесь по смыслу больше подходит assertNull, а не assertNotEquals.

yajaneya committed:
Добавил проверки на null (что пользователя нет в БД) перед созданием, а перед удалением - проверки на его присутствие в БД.

yajaneya committed:
заменил assertEquals (null,...) на assertNull(...)

yajaneya committed:
добавил File.separator в HibernateUtil

yajaneya committed:
зачистил pom.xml
